### ISSUE TITLE
Legend hiding rules a bit too strict?

### ISSUE THREAD

**Issue by cpsievert** [ropensci/plotly#212]
I discovered [this example](http://stackoverflow.com/questions/17521438/geom-rect-and-alpha-does-this-work-with-hard-coded-values) via the checkmark on #161 titled "geom_rect()"

This works fine:

```r
base <- ggplot(mtcars) + 
  geom_density(aes(x=disp, group=cyl, fill=factor(cyl)), alpha=0.6, adjust=0.75) 
```

![](https://plot.ly/~cpsievert/532/density-vs-disp.png)

But once I add a "rect layer", the legend goes away (it shouldn't):

```r
base + annotate("rect", xmin=100, xmax=200, ymin=0, ymax=0.02, alpha=0.2, fill="red")
```

![](https://plot.ly/~cpsievert/531/density-vs-disp.png)

**Comment by 13bzhang** [comment]
Now that you mentioned it in #225, I shall take a look.

**Comment by 13bzhang** [comment]
Interesting. When I ran the same code as you for the first plot, I did not get the legends -- could be a change in the code since April 30.

![density-vs-disp_bad](https://cloud.githubusercontent.com/assets/6131842/8076593/fe482d34-0f1b-11e5-892a-b04df916d37f.png)

**Comment by 13bzhang** [comment]
Legends have been turned off for boxplots as well. At least they are not off for barcharts.

```r
m <- ggplot(mtcars, aes(factor(cyl), mpg))
p <- m + geom_boxplot(aes(fill = factor(cyl)))
py$ggplotly(p)
```

ggplot2

![boxplot_good](https://cloud.githubusercontent.com/assets/6131842/8076680/0147c368-0f1d-11e5-8e53-c9352f86ab7e.png)

plotly

![boxplot_bad](https://cloud.githubusercontent.com/assets/6131842/8076683/0b71e968-0f1d-11e5-9508-d8cc990ffd00.png)

**Comment by cpsievert** [comment]
Fixed via 5509909bc6bec2251691ec552f541ae0b1dd9e9e
